### ISSUE TITLE
Support parsing Snowflake style DATEADD() function with date part as string literal

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#11938]
This is valid Snowflake SQL:

```sql
-- With a string literal
select dateadd('month', 1, date '2000-01-01')

-- With a date part
select dateadd(month, 1, date '2000-01-01')
```

We currently only support the date part version.

**Comment by lukaseder** [comment]
See also: https://github.com/jOOQ/jOOQ/issues/11931

**Comment by lukaseder** [comment]
It seems that all date parts can be used as string literals in Snowflake, e.g. also:

```sql
select extract('month' from date '2000-01-01')
```

**Comment by lukaseder** [comment]
Fixed for jOOQ 3.15.0
